testreporters/log_scanner.go

```diff
@@ -0,0 +1,84 @@
+package testreporters
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/rs/zerolog"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	OneLogAtLogLevelErr       = "found log at level"
+	MultipleLogsAtLogLevelErr = "found too many logs at level"
+)
+
+// ScanLogLine scans a log line for a failing log level, returning the number of failing logs found so far. It return an error if the failure threshold is reached or if any panic is found
+// or if there's no log level found. It also takes a list of allowed messages that are ignored if found.
```

Comment:
Can I not specify both a log level to check for, and message to exclude from worrying about?

Reply:
wdym?

Reply:
NVM, didn't see the period